x-pack/plugins/alerting_builtins/server/alert_types/index_threshold/action_context.ts

```diff
@@ -65,5 +65,5 @@ export function addMessages(
     }
   );
 
-  return { ...baseContext, title, message };
+  return { ...params, ...baseContext, title, message };
```

Comment:
Seems like what we should be doing is adding a new top-level "variable" for mustache called `params`, for the alert params.  That way we'll get a nice grouping, otherwise the values calculated by the executor will be mixed in with the parameters, if they're all under `context`.  And EVERY alert type will end up being able provide these for free.

I'm not quite sure how we'd populate the list of variables underneath it, for the UI popup with the list of variables / description.  I don't think the description is probably all that important to have a description, but I'm not even sure how we'd get the list of top-level properties of the alert params for an alert type.  Without hard-coding it somehow.  I don't think there's anything in config-schema to help us out there, but not sure.



Idea would be to add `params` to the list of other mustache variables here:

https://github.com/elastic/kibana/blob/master/x-pack/plugins/alerts/server/task_runner/transform_action_params.ts#L22-L31

and then get `params` from the `ExecutionHandlerOptions` here and send into the `transformActionParams()` function: https://github.com/elastic/kibana/blob/master/x-pack/plugins/alerts/server/task_runner/create_execution_handler.ts#L50-L71